From e191ba2f8ba242595d7941678f3f5a0441cfe776 Mon Sep 17 00:00:00 2001
From: David Redondo <qt@david-redondo.de>
Date: Wed, 7 Oct 2026 21:00:37 +0200
Subject: [PATCH] QDesktopUnixServices: Add missing cc and bcc fields to portal
 call

mailto URIs can specify cc and bcc, so far we were just ignoring these.

Change-Id: I170d48f72f89f9dcafe051562c98fa9231a840d0
Reviewed-by: Liang Qi <liang.qi@qt.io>
(cherry picked from commit 0097d9d830a48c55c613cfcb8946602ee8c76652)
Reviewed-by: David Redondo <qt@david-redondo.de>
Origin: upstream, https://invent.kde.org/qt/qt/qtbase/-/commit/e191ba2f8ba242595d7941678f3f5a0441cfe776
Forwarded: not-needed
Last-Update: 2026-10-29
---
 src/gui/platform/unix/qdesktopunixservices.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/gui/platform/unix/qdesktopunixservices.cpp b/src/gui/platform/unix/qdesktopunixservices.cpp
index 1883cd295aa..9b1ed60a6b1 100644
--- a/src/gui/platform/unix/qdesktopunixservices.cpp
+++ b/src/gui/platform/unix/qdesktopunixservices.cpp
@@ -267,6 +267,10 @@ static inline QDBusMessage xdgDesktopPortalSendEmail(const QUrl &url, const QStr
         options.insert("subject"_L1, subject);
     if (const auto body = urlQuery.queryItemValue("body"_L1); !body.isEmpty())
         options.insert("body"_L1, body);
+    if (const auto cc = urlQuery.allQueryItemValues("cc"_L1, QUrl::FullyDecoded); !cc.isEmpty())
+        options.insert("cc"_L1, cc);
+    if (const auto bcc = urlQuery.allQueryItemValues("bcc"_L1, QUrl::FullyDecoded); !bcc.isEmpty())
+        options.insert("bcc"_L1, bcc);
 
     // O_PATH seems to be present since Linux 2.6.39, which is not case of RHEL 6
 #ifdef O_PATH
